src/Features/Core/Portable/UseSystemHashCode/Analyzer.cs

```diff
@@ -0,0 +1,352 @@
+                    if (binary.RightOperand.IsNullLiteral() &&
+                        TryGetFieldOrProperty(binary.LeftOperand, out _))
+                    {
+                        if (binary.OperatorKind == BinaryOperatorKind.Equals)
+                        {
+                            // (StringProperty == null ? 0 : StringProperty.GetHashCode())
+                            return TryAddHashedSymbol(conditional.WhenFalse, seenHash: true);
+                        }
+                        else if (binary.OperatorKind == BinaryOperatorKind.NotEquals)
+                        {
+                            // (StringProperty != null ? StringProperty.GetHashCode() : 0)
+                            return TryAddHashedSymbol(conditional.WhenTrue, seenHash: true);
+                        }
+                    }
+                }
+
+                // Look to see if we're referencing some field/prop/base.  However, we only accept
+                // this reference if we've at least been through something that indicates that we've
+                // hashed the value.
+                if (seenHash)
+                {
+                    if (value is IInstanceReferenceOperation instanceReference)
+                    {
+                        // reference to this/base.
+                        //
+                        // Happens with code like: `var hashCode = base.GetHashCode();`
+                        return instanceReference.ReferenceKind == InstanceReferenceKind.ContainingTypeInstance;
+                    }
+
+                    // After decomposing all of the above patterns, we must end up with an operation that is
+                    // a reference to an instance-field (or prop) in our type.  If so, and this is the only
+                    // time we've seen that field/prop, then we're good.
+                    //
+                    // We only do this if we actually did something that counts as hashing along the way.  This
+                    // way
+                    if (TryGetFieldOrProperty(value, out var fieldOrProp) &&
+                        Equals(fieldOrProp.ContainingType.OriginalDefinition, method.ContainingType))
+                    {
+                        return Add(hashedSymbols, fieldOrProp);
+                    }
+                }
+
+                // Anything else is not recognized.
+                return false;
+            }
+        }
+
+        private static bool TryGetFieldOrProperty(IOperation operation, out ISymbol symbol)
+        {
+            if (operation is IFieldReferenceOperation fieldReference)
+            {
+                symbol = fieldReference.Member;
+                return !symbol.IsStatic;
+            }
+
+            if (operation is IPropertyReferenceOperation propertyReference)
+            {
+                symbol = propertyReference.Member;
+                return !symbol.IsStatic;
+            }
+
+            symbol = null;
+            return false;
```

Comment:
Could be:

```cs
symbol = (operation as IFieldReferenceOperation)?.Member ??
         (operation as IPropertyReferenceOperation)?.Member;
return symbol is { IsStatic: false };
```

Reply:
this is cool! :)  Though the `??` doesn't work since the two symbol types are not compatible. :(

Reply:
Ah, you can always cast one to `ISymbol` though that makes it less neat.